### PR TITLE
Service graphs documentation

### PR DESCRIPTION
**Pull request by mapno**
**What this PR does**:

Adds documentation for _Service Graphs_.

The images added are the following:

image

service-graphs-agent-architecture

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+The Agent will process traces and generate service graphs in the form of prometheus metrics.
+The generated metrics represent edges between nodes in the graph. Nodes are represented by `client` and `server` labels.
+
+```
+  tempo_service_graph_request_total{client="app", server="db"} 20
+```
+
+<p align="center"><img src="../service-graphs-agent-architecture.png" alt="Service graphs architecture"></p>
+
+Service graphs works by inspecting spans and looking for the tag `span.kind`.
```
[the remark this answers is not in the document]
Comment (mapno) [reply docs/tempo/website/grafana-agent/service-graphs.md]: Added

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+```
+traces:
+  configs:
+    - name: default
+      ...
+      service_graphs:
+        enabled: true
+```
+
+To see all the available config options, refer to the [configuration reference](https://github.com/grafana/agent/blob/main/docs/configuration/traces-config.md).
```
[the remark this answers is not in the document]
Comment (mapno) [reply docs/tempo/website/grafana-agent/service-graphs.md]: Yeah, good suggestion. Added

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    - name: default
+      scrape_configs:
+        - job_name: local_scrape
+          static_configs:
+            - targets: ['127.0.0.1:12345']
+      remote_write:
+        - url: <remote_write>
+```
+
+### Grafana
```
[the remark this answers is not in the document]
Comment (mapno) [reply docs/tempo/website/grafana-agent/service-graphs.md:88]: Good idea. I added a minimal example. Thanks

**Review by kvrhdn — APPROVED** [review]
This looks good to me, the technical details look right :+1: I think a review from @KMiller-Grafana will be very valuable as well before merging.

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+## Cardinality
+
+Cardinality can pose a problem when you have lots of services.
+There isn't a direct formula or solution to this problem.
+But the following guide should help estimate the cardinality that the feature will generate.
+
+### How to estimate the cardinality
+
+The amount of edges depends on the amount of nodes in the system and the direction of the requests between them.
+Let’s call this amount hops. Every hop will be a unique combination of client + server labels.
```
Comment (KMiller-Grafana) [inline docs/tempo/website/grafana-agent/service-graphs.md:129]:
```suggestion
There isn't a direct formula or solution to the cardinality problem,
but the following guide should help estimate the cardinality that the feature will generate.

The quantity of graph edges depend on both the quantity of nodes in the distributed system and the direction of the requests between the nodes.
Call the quantity of edges to be hops. Every hop will be a unique combination of client plus server labels.
```

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+For example:
+- a system with 3 nodes `(A, B, C)` of which A only calls B and B only calls C will have 2 hops `(A → B, B → C)`
+- a system with 3 nodes `(A, B, C)` that call each other (i.e. all bidirectional links somehow) will have 6 hops `(A → B, B → A, B → C, C → B, A → C, C → A)`
+
+We can’t calculate the amount of hops automatically based upon the nodes,
+but it should be a value between `#services - 1` and `#services!`.
+
+If we know the amount of hops in a system, we can calculate the cardinality of the generated service graphs:
```
Comment (KMiller-Grafana) [inline docs/tempo/website/grafana-agent/service-graphs.md:138]:
```suggestion
- a fully-connected system with 3 nodes `(A, B, C)` will have 6 hops `(A → B, B → A, B → C, C → B, A → C, C → A)`

We can’t calculate hops automatically based upon the nodes,
but it should be a value between the `number of services - 1` and  `the number of services!`.

If we know the hops in a system, we can calculate the cardinality of the generated service graph:
```

File: docs/tempo/website/grafana-agent/service-graphs.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+```
+  service_graph_request_total: #hops
+  service_graph_request_failed_total: #hops
+  service_graph_request_server_seconds: 3 buckets * #hops
+  service_graph_request_client_seconds: 3 buckets * #hops
+  service_graph_unpaired_spans_total: #services (absolute worst case)
+  service_graph_dropped_spans_total: #services (absolute worst case)
+```
+
+Finally, we get the following cardinality estimation:
```
Comment (KMiller-Grafana) [inline docs/tempo/website/grafana-agent/service-graphs.md:149]:
```suggestion
  service_graph_request_total: number of hops
  service_graph_request_failed_total: number of hops
  service_graph_request_server_seconds: 3 buckets * number of hops
  service_graph_request_client_seconds: 3 buckets * number of hops
  service_graph_unpaired_spans_total: number of services (absolute worst case)
  service_graph_dropped_spans_total: number of services (absolute worst case)
```

The cardinality estimation becomes:
```

File: docs/tempo/website/grafana-agent/tail-based-sampling.md [hunk 7]
```diff
@@ -43,10 +43,10 @@ This overhead will increase with the number of Agent instances that share the sa
 To start using tail-based sampling, define a sampling policy.
 If you're using a multi-instance deployment of the agent,
 add load balancing and specify the resolving mechanism to find other Agents in the setup.
-To see all the available config options, refer to the [configuration reference](https://github.com/grafana/agent/blob/main/docs/configuration/tempo-config.md).
+To see all the available config options, refer to the [configuration reference](https://grafana.com/docs/agent/latest/configuration/traces-config/).
```
Comment (KMiller-Grafana) [inline docs/tempo/website/grafana-agent/tail-based-sampling.md:46]:
```suggestion
To see all the available configuration options, refer to the [traces_config](https://grafana.com/docs/agent/latest/configuration/traces-config/) configuration reference.
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
Lots of improvement suggestions!

**Review by KMiller-Grafana — APPROVED** [review]
Thanks for all the changes.  LGTM.

**Pull request merged**